pkg/resourceinterpreter/default/native/aggregatestatus.go

```diff
@@ -279,39 +280,44 @@ func aggregateDaemonSetStatus(object *unstructured.Unstructured, aggregatedStatu
 
-		// always set 'observedGeneration' with current generation(.metadata.generation)
-		// which is the generation Karmada 'observed'.
-		// The 'observedGeneration' is mainly used by GitOps tools(like 'Argo CD') to assess the health status.
-		// For more details, please refer to https://argo-cd.readthedocs.io/en/stable/operator-manual/health/.
+		// `memberStatus.ObservedGeneration >= memberStatus.Generation` means the member's status corresponds the latest spec revision of the member DaemonSet.
+		// `memberStatus.ResourceTemplateGeneration >= daemonSet.Generation` means the member DaemonSet has been aligned with the latest spec revision of federated DaemonSet.
+		// If both conditions are met, we consider the member's status corresponds the latest spec revision of federated DaemonSet.
+		if member.ObservedGeneration >= member.Generation &&
+			member.ResourceTemplateGeneration >= daemonSet.Generation {
+			observedLatestResourceTemplateGenerationCount++
+		}
+
+		newStatus.CurrentNumberScheduled += member.CurrentNumberScheduled
+		newStatus.DesiredNumberScheduled += member.DesiredNumberScheduled
+		newStatus.NumberAvailable += member.NumberAvailable
+		newStatus.NumberMisscheduled += member.NumberMisscheduled
+		newStatus.NumberReady += member.NumberReady
+		newStatus.UpdatedNumberScheduled += member.UpdatedNumberScheduled
+		newStatus.NumberUnavailable += member.NumberUnavailable
+	}
+
+	// The 'observedGeneration' is mainly used by GitOps tools(like 'Argo CD') to assess the health status.
+	// For more details, please refer to https://argo-cd.readthedocs.io/en/stable/operator-manual/health/.
+	if observedLatestResourceTemplateGenerationCount == len(aggregatedStatusItems) {
 		newStatus.ObservedGeneration = daemonSet.Generation
-		newStatus.CurrentNumberScheduled += temp.CurrentNumberScheduled
-		newStatus.DesiredNumberScheduled += temp.DesiredNumberScheduled
-		newStatus.NumberAvailable += temp.NumberAvailable
-		newStatus.NumberMisscheduled += temp.NumberMisscheduled
-		newStatus.NumberReady += temp.NumberReady
-		newStatus.UpdatedNumberScheduled += temp.UpdatedNumberScheduled
-		newStatus.NumberUnavailable += temp.NumberUnavailable
+	} else {
+		newStatus.ObservedGeneration = oldStatus.ObservedGeneration
 	}
 
-	if oldStatus.ObservedGeneration == newStatus.ObservedGeneration &&
-		oldStatus.CurrentNumberScheduled == newStatus.CurrentNumberScheduled &&
-		oldStatus.DesiredNumberScheduled == newStatus.DesiredNumberScheduled &&
-		oldStatus.NumberAvailable == newStatus.NumberAvailable &&
-		oldStatus.NumberMisscheduled == newStatus.NumberMisscheduled &&
-		oldStatus.NumberReady == newStatus.NumberReady &&
-		oldStatus.UpdatedNumberScheduled == newStatus.UpdatedNumberScheduled &&
-		oldStatus.NumberUnavailable == newStatus.NumberUnavailable {
+	if equality.Semantic.DeepEqual(oldStatus, newStatus) {
```

Comment:
Other fields in the status may affect the current judgment.

Reply:
other fields should be empty.

Reply:
If other fields are added in the future, such as condition, the judgment here will be invalid.

Reply:
I don't think there will be this problem. even if a new field is added, DaemonSet of Karmada control plane will have this field at the end, and this judgment is still legal.

Reply:
Does that mean that other types of resources can be treated the same way?

Reply:
yes, but I only modify this because of a lint error: cyclomatic complexity too high :joy:

Reply:
okay
/lgtm
/cc @yike21 @veophi

Reply:
thanks!
/lgtm